src/main/java/seiki/ui/Ui.java

```diff
@@ -0,0 +1,144 @@
+package seiki.ui;
+
+import static seiki.common.Messages.MESSAGE_ADD_SUCCESS;
+import static seiki.common.Messages.MESSAGE_DELETE_SUCCESS;
+import static seiki.common.Messages.MESSAGE_FAREWELL;
+import static seiki.common.Messages.MESSAGE_FIND_SUCCESS;
+import static seiki.common.Messages.MESSAGE_GREETING;
+import static seiki.common.Messages.MESSAGE_LIST_SUCCESS;
+import static seiki.common.Messages.MESSAGE_LOGO;
+import static seiki.common.Messages.MESSAGE_MARK_SUCCESS;
+import static seiki.common.Messages.MESSAGE_REMAINING_TASKS;
+import static seiki.common.Messages.MESSAGE_TASK;
+import static seiki.common.Messages.MESSAGE_UNMARK_SUCCESS;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+
+import seiki.data.TaskList;
+import seiki.data.task.Task;
+
+/**
+ * User Interface of the chatbot.
+ */
+public class Ui {
+
+    private static final String DIVIDER = "────────────────────────────────────────────────────────────";
+
+    private final Scanner in;
+    private final PrintStream out;
+    public Ui() {
+        this(System.in, System.out);
+    }
+
+    /**
+     * Initializes the input and output of the chatbot.
+     * @param in
+     * @param out
+     */
+    public Ui(InputStream in, PrintStream out) {
+        this.in = new Scanner(in);
+        this.out = out;
+    }
+
+    /**
+     * Generates and prints the welcome message upon the start of the chatbot.
+     */
+    public void showWelcome() {
+        showToUser(DIVIDER,
+                MESSAGE_LOGO,
+                MESSAGE_GREETING,
+                DIVIDER);
+    }
+
+    /**
+     * Shows the message(s) to the user.
+     * @param message
+     */
+    public void showToUser(String... message) {
+        for (String m : message) {
+            out.println(m);
+        }
+    }
+
+    public String readCommand() {
+        return in.nextLine();
+    }
+
+    public void showLine() {
+        showToUser(DIVIDER);
+    }
+
+    public void showError(String message) {
+        showToUser(message);
+    }
```

Comment:
Javadocs would be appreciated for these methods

Reply:
Okay, I will add them later on.